src/MessageHandler/ActivityPub/Inbox/ActivityHandler.php

```diff
@@ -94,14 +94,18 @@ private function handle(?array $payload)
 
         if ('Announce' === $payload['type']) {
             $actor = $this->manager->findActorOrCreate($payload['actor']);
-            if ($actor instanceof Magazine) {
-                $actor->lastOriginUpdate = new \DateTime();
-                $this->entityManager->persist($actor);
-                $this->entityManager->flush();
-            }
+
             // we check for an array here, because boosts are announces with an url (string) as the object
             if (\is_array($payload['object'])) {
                 $payload = $payload['object'];
+
+                if ('Create' === $payload['type'] && $actor instanceof Magazine && $actor->lastOriginUpdate < (new \DateTime())->modify('-3 hours')) {
+                    // announced creates update magazine last origin timestamp when after threshold
+                    $actor->lastOriginUpdate = new \DateTime();
```

Comment:
Could you rename this variable? I was confused whether this works, because this is not the actor of the `$payload` variable at this position.
```suggestion
                    $announceActor->lastOriginUpdate = new \DateTime();
```